platforms/openid-connect/README.md

```diff
@@ -0,0 +1,110 @@
+## Unleash Community Docker Image with OpenID Connect Support
+
+**Useful links:**
+
+- [Docker image on dockerhub](https://hub.docker.com/r/unleashorg/unleash-community-openid-connect)
+- [passport-openid-connect](https://www.passportjs.org/packages/passport-openidconnect/)
+
+For an example usage see the example below.
+You can also visit the [passport](https://www.passportjs.org/docs/) and [openid-connect](https://www.passportjs.org/packages/passport-openidconnect/) documentations for further information.
+
+## Example
+
+This shows the usage of the OpenIdConnectStrategy with a keycloak realm as IDP.
+It authenticates the user and checks some custom roles to give the user some initial permissions. 
+It also updates the role in case it changed, so you can completely control unleash permissions through roles in keycloak.
+You need to configure your client accordingly, so it returns roles as a `resource_access` object.
+
+If you only need the authentication, just remove that part.
+
+```javascript
+const unleash = require('unleash-server')
+
+function openIdConnect(app, config, services) {
+    const {userService} = services;
+
+    passport.use(new OpenIdConnectStrategy({
+        issuer: `${host}/realms/${realm}`,
+        authorizationURL: `${host}/realms/${realm}/protocol/openid-connect/auth`,
+        tokenURL: `${host}/realms/${realm}/protocol/openid-connect/token`,
+        userInfoURL: `${host}/realms/${realm}/protocol/openid-connect/userinfo`,
+        clientID: clientID,
+        clientSecret: '',
+        callbackURL: `${contextPath}/api/auth/callback`,
+        skipUserProfile: false
```

Comment:
My keycloak instance did not allow me to retrieve the UserProfile:

```
unleash       | err InternalOAuthError: Failed to obtain access token
unleash       |     at /unleash/node_modules/passport-openidconnect/lib/strategy.js:143:29
unleash       |     at /unleash/node_modules/oauth/lib/oauth2.js:191:18
unleash       |     at ClientRequest.<anonymous> (/unleash/node_modules/oauth/lib/oauth2.js:162:5)
unleash       |     at ClientRequest.emit (events.js:400:28)
unleash       |     at Socket.socketErrorListener (_http_client.js:475:9)
unleash       |     at Socket.emit (events.js:400:28)
unleash       |     at emitErrorNT (internal/streams/destroy.js:106:8)
unleash       |     at emitErrorCloseNT (internal/streams/destroy.js:74:3)
unleash       |     at processTicksAndRejections (internal/process/task_queues.js:82:21) {
unleash       |   oauthError: Error: connect ECONNREFUSED 192.168.96.5:80
unleash       |       at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16) {
unleash       |     errno: -111,
unleash       |     code: 'ECONNREFUSED',
unleash       |     syscall: 'connect',
unleash       |     address: '192.168.96.5',
unleash       |     port: 80
unleash       |   }
unleash       | }
```

I am curious is there some special setting within keycloak that needs to be set? Or was it because I was lacking the roles information? 🤔

Reply:
Did you get it working? From the trace I can see that you had general problems with connecting to your keycloak instance.
I did not set any special settings, aside from the roles and the default openid stuff like allowed redirect urls, etc.

Reply:
Yes I got it working, but it was late and I pasted a wrong error. 🙈  I was getting the error: `InternalOAuthError: failed to fetch user profile`

Therefore needed to take out the roles logic and turn `skipUserProfile: false` to `skipUserProfile: true`. This way I was able to login without problems.

Reply:
Could also be a problem with an old version of keycloak I was running

Reply:
> Therefore needed to take out the roles logic and turn skipUserProfile: false to skipUserProfile: true. This way I was able to login without problems.

Ah that makes sense. The roles are not included in the id token by default. You need to add them with a mapper, like this:

![image](https://user-images.githubusercontent.com/465359/220576992-1110495e-f10d-4fe0-959a-12d75471b21f.png)